### ISSUE TITLE
Can't create unknown regex: $RE{comment}{C++}

### ISSUE THREAD

**Issue by kkettinger** [AlDanial/cloc#745]
**Describe the bug**
We are experiencing an error with cloc that we can't track down. It happens only very sporadically in our yocto environment and we get the following error message from cloc:

```
Can't create unknown regex: $RE{comment}{C++} at /usr/bin/cloc line 10842.
	...propagated at /usr/bin/cloc line 6528.
```

**cloc; OS; OS version**
 - cloc version: 1.92
 - If running the cloc source, Perl version: v5.34.1
 - OS (eg Linux, Windows, macOS, etc): Linux Ubuntu
 - OS version: 22.04

**To Reproduce**
Can't reproduce at the moment.

I appreciate any help on this issue.

**Comment by AlDanial** [comment]
The Ubuntu version of cloc uses the locally installed version of the Regexp::Common module rather than the one bundled with cloc.  Possibly the Regexp::Common module you have installed is messed up somehow?  Try using the vanilla cloc code pulled from the Releases section.

**Comment by kkettinger** [comment]
We are using cloc from the release section (bundled with meta-oe, see https://layers.openembedded.org/layerindex/recipe/117971/). The regexp module is included with version `Regexp::Common v2017060201`.
I compared the hashes of the `cloc` file just to be sure it is the same as from the release section.

We are calling cloc with these arguments:
```
cloc
    --include-lang=C++,'C/C++ Header' \
    --force-lang-def=cloc_definitions.txt \
    --by-file \
    --fullpath \
    --not-match-d='(test)' \
    --json \
    --out=cloc.json
```

I've found two issues with the same error message:
- https://github.com/CesiumGS/cesium/issues/6224
- https://github.com/AlDanial/cloc/issues/49

But with not apparent solution or reason why this error message happens.
